src/main/kotlin/com/sourcegraph/cody/agent/CodyAgent.kt

```diff
@@ -156,12 +159,27 @@ private constructor(
         processBuilder.environment()["CODY_LOG_EVENT_MODE"] = "connected-instance-only"
       }
 
+      if (ConfigUtil.isIntegrationTestModeEnabled()) {
+        // N.B. Do not set CODY_TESTING=true -- that is for Agent-side tests.
+        val testToken = System.getenv("CODY_INTEGRATION_TEST_TOKEN")
```

Comment:
I think we should reuse tokens and polly.js recording infrastructure we already have in the agent.
This way we won't need to confugure/understand/setup two independent testing environments.

Reply:
Some test conditions are hard to recreate with production backends. VSCode uses both kinds to good effect.